core/src/main/java/com/linecorp/armeria/client/retry/RetryingHttpClient.java

```diff
@@ -169,13 +169,15 @@ private void retry0(int currentAttemptNo, Backoff backoff, ClientRequestContext
                         DefaultHttpResponse rootResponse, Throwable exception, long millisAfter) {
         long nextDelay = backoff.nextDelayMillis(currentAttemptNo);
         if (nextDelay < 0) {
+            rootReqDuplicator.close();
             rootResponse.close(exception);
         } else {
             final EventLoop eventLoop = ctx.contextAwareEventLoop();
             nextDelay = Math.max(nextDelay, millisAfter);
             try {
                 nextDelay = getNextDelay(nextDelay, ctx);
             } catch (ResponseTimeoutException e) {
+                rootReqDuplicator.close();
                 rootResponse.close(e);
```

Comment:
How about closing `rootResponse` before closing the duplicator? #667 will make `duplicator.close()` abort the root stream.